ivy.el

```diff
@@ -3227,12 +3227,20 @@ Otherwise, ~/ will move home."
 
 (defvar ivy--exhibit-timer nil)
 
+(let (last-pos)
```

Comment:
I recommend either turning this into a `defvar`, or defining named closues like so:

```el
(defalias 'foo
  (let (...)
    (lambda (...) ...))
  "...")
```
since Emacs does not always handle `defun`-within-`let` well; see for example the following Emacs SE questions:
* https://emacs.stackexchange.com/q/29853/15748
* https://emacs.stackexchange.com/q/32628/15748

There's also an aesthetic argument for keeping only definitions at top-level.

Reply:
In this case, I would still stick to closures, as I thought every little helps for namespace pollution, if it can be avoided, thanks!

Reply:
No need to worry about one more internal (with double `--` hyphens in the name) variable w.r.t. namespace pollution.